### DISCUSSION TITLE
Torrents marked as finished despite ratio group action to 'continue'

### DISCUSSION THREAD

**Discussion by Span24** [rakshasa/rtorrent#1196] [General]
I have my ratio group actions set to continue seeding despite the ratio.  Nonetheless, once a seeded torrent tops 1:1 its marked as finished and seeding stops.  I'm running multiple instances of v3.10 and all are configured exactly the same.  Two are seedboxes and one a local unraid docker container.  Only one of them (the local instance) is having this 'problem'.

What could be the cause?

**Comment by kannibalox** [comment]
rTorrent's latest version is 0.9.8, you may be looking for https://github.com/Novik/ruTorrent instead

**Reply by Span24** [reply]
Thank you!  But I am uncertain which is responsible for these actions.

**Reply by kannibalox** [reply]
The easiest way to narrow it down is to see if it still happens while ruTorrent is turned off. Can you provide the rtorrent.rc configuration you're using to set the group actions?

**Comment by Span24** [comment]
Im not so certain that the group actions config is in rtorrent.rc but stored in a ratio.dat and/or ratiorules.dat file.  I'll include my rtorrent.rc for your review.
[rtorrent.rc.txt](https://github.com/rakshasa/rtorrent/files/10188577/rtorrent.rc.txt)

**Reply by kannibalox** [reply]
The only other thing I can think of is that a group was created on the local and then deleted, but the ratio group settings still exist in rTorrent and that's what's triggering the stop. I don't use ruTorrent, so maybe there's a cleaner way to do this, but if you stop both rTorrent and ruTorrent, remove ratio.dat and ratiorules.dat, and re-set the rules after both are started again, that would fix it.

**Comment by Span24** [comment]
After weeks of researching and testing this I seem to have found the reason for the problem!  A watchdog script used in the docker container Im using (binhex: arch-rtorrentvpn) restarts rtorrent whenever there is a problem with the vpn.  Part of this restart reinitializes the plugins with the following line in rtorrent.rc and /home/username/initplugins.sh:
execute = {/bin/bash,-c,/usr/bin/sleep 10s && /usr/bin/php /usr/share/webapps/rutorrent/php/initplugins.php $WEBUI_USER &}

This 'reset' appears to have the effect of causing any torrents with a ratio of 1:1 or better to stop seeding regardless of configuration.  The remarks in the files indicate that it is there so that the plugins 'rss' and 'scheduler' will work properly.  I do not use these rutorrent plugins so I disabled them and commented out the line in both files.  So far (48 hours) I have not seen a problem and my initial problem is GONE!

This is a great docker image and i sincerely hope this saves someone else the frustration I experienced.

In any case, this is NOT the fault of rtorrent but the watchdog script included with the image.  I thank you for your assistance!

**Reply by kannibalox** [reply]
Glad to hear you figured it out! Thanks for sharing your solution.
